### PR TITLE
Support Ellipse Shape for InstSeg algo

### PR DESCRIPTION
**Pull request by eugene123tw**
### Summary

Fix https://jira.devtools.intel.com/browse/CVS-159219 by supporting Ellipse shape

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have ran e2e tests and there is no issues.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2024 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Review by sovrasov — DISMISSED** [review]

**Comment by sovrasov** [comment]
@eugene123tw if you have a compact arrow to test the changes, please save it for later: we have to think about testing on arrow datasets on OTX CI in the future.

**Review by kprokofi — DISMISSED** [review]

File: src/otx/core/data/dataset/instance_segmentation.py [hunk 1]
```diff
@@ -42,23 +44,49 @@ def _get_item_impl(self, index: int) -> InstanceSegDataEntity | None:
...
+            anno_collection[anno.__class__.__name__].append(anno)
+
         gt_bboxes, gt_labels, gt_masks, gt_polygons = [], [], [], []
 
-        for annotation in item.annotations:
-            if isinstance(annotation, Polygon):
-                bbox = np.array(annotation.get_bbox(), dtype=np.float32)
+        # NOTE (Eugene):
+        # Temporary solution to handle multiple annotation types.
+        # Ideally, we should pre-filter annotations during initialization of the dataset.
```
Comment (Daankrol) [inline src/otx/core/data/dataset/instance_segmentation.py]: Should we make a ticket/issue for this to make this change in the future and add a reference here?

File: src/otx/core/data/dataset/tile.py [hunk 2]
```diff
@@ -467,26 +508,49 @@ def _get_item_impl(self, index: int) -> TileInstSegDataEntity:  # type: ignore[o
...
+            anno_collection[anno.__class__.__name__].append(anno)
+
         gt_bboxes, gt_labels, gt_masks, gt_polygons = [], [], [], []
 
-        for annotation in item.annotations:
-            if isinstance(annotation, Polygon):
-                bbox = np.array(annotation.get_bbox(), dtype=np.float32)
+        # NOTE (Eugene):
+        # Temporary solution to handle multiple annotation types.
+        # Ideally, we should pre-filter annotations during initialization of the dataset.
```
Comment (Daankrol) [inline src/otx/core/data/dataset/tile.py]: again, maybe add a reference to issue/ticket for future reference.

**Review by Daankrol — DISMISSED** [review]
Can be merged if tests are successful.

File: src/otx/core/data/dataset/tile.py (hunk 2 above)
Comment (eugene123tw) [reply src/otx/core/data/dataset/tile.py]: Good point! I added a Jira ticket tracking the issue for future work.

File: src/otx/core/data/dataset/instance_segmentation.py (hunk 1 above)
Comment (eugene123tw) [reply src/otx/core/data/dataset/instance_segmentation.py]: Good point! I added a Jira ticket tracking the issue for future work.

**Review by ashwinvaidya17 — APPROVED** [review]

**Review by sovrasov — APPROVED** [review]

**Pull request merged**